### PR TITLE
bump up to Serde M4

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ bcpkix = "1.70"
 micronaut-test = "4.0.0-M3"
 micronaut-multitenancy = "5.0.0-M2"
 micronaut-reactor = "3.0.0-M1"
-micronaut-serde = "2.0.0-M1"
+micronaut-serde = "2.0.0-M4"
```
Comment (sdelamo) [inline gradle/libs.versions.toml:19]: Serde M4 includes https://github.com/micronaut-projects/micronaut-serialization/pull/439 Thanks for the quick fix @yawkat

File: security-oauth2/src/test/groovy/io/micronaut/security/oauth2/client/clientcredentials/ClientCredentialsConcurrentSpec.groovy [hunk 2]
```diff
@@ -156,22 +159,54 @@ class ClientCredentialsConcurrentSpec extends Specification {
             this.tokenExpiration = tokenExpiration
         }
 
+
+        @Introspected
+        static class ClientCredentialsForm {
```
Comment (sdelamo) [inline security-oauth2/src/test/groovy/io/micronaut/security/oauth2/client/clientcredentials/ClientCredentialsConcurrentSpec.groovy:164]: I had to extract this POJOs to make the test pass with serde 4

File: security-oauth2/src/test/groovy/io/micronaut/security/oauth2/client/clientcredentials/ClientCredentialsConcurrentSpec.groovy [hunk 3]
```diff
@@ -97,9 +98,11 @@ class ClientCredentialsConcurrentSpec extends Specification {
     @AutoCleanup
     BlockingHttpClient client = httpClient.toBlocking()
 
+    @IgnoreIf({ (Runtime.runtime.availableProcessors().intdiv(2) ?: 1) == 1 })
     void "no exception for concurrent requests using client credentials"() {
         when:
-        int numberOfFutures = Runtime.getRuntime().availableProcessors() - 1
+        int numberOfFutures = Runtime.runtime.availableProcessors().intdiv(2) ?: 1
```
Comment (sdelamo) [inline security-oauth2/src/test/groovy/io/micronaut/security/oauth2/client/clientcredentials/ClientCredentialsConcurrentSpec.groovy:104]: I am borrowing a page from [Gradle Execute tests in parallel documentation](https://docs.gradle.org/current/userguide/performance.html#execute_tests_in_parallel). This makes this test pass when running in parallel. I have a 10 machine and `Runtime.runtime.availableProcessors()` returns 20.

**Review by timyates — APPROVED** [review]

File: security-oauth2/src/test/groovy/io/micronaut/security/oauth2/client/clientcredentials/ClientCredentialsConcurrentSpec.groovy (hunk 2 above)
Comment (sdelamo) [reply security-oauth2/src/test/groovy/io/micronaut/security/oauth2/client/clientcredentials/ClientCredentialsConcurrentSpec.groovy:164]: I created an issue in serde https://github.com/micronaut-projects/micronaut-serialization/issues/441

**Pull request merged**